### ISSUE TITLE
Do you have plan to support vulkan to improve performance?

### ISSUE THREAD

**Issue by guoguo338** [makepad/makepad#511]

**Comment by makepaddev** [comment]
We do, its on our todo list relatively soon.

**Comment by guoguo338** [comment]
Great, do you expect GUI performance gains from that? And is there any other plans to improve performance?

**Comment by guofoo** [comment]
Yes, there will be GUI performance gains from adding support to Vulkan. There are many continued work and plans to improve performance such as using system fonts to reduce startup time and memory footprint, improved rendering, etc. 

Even as it is now, Makepad performance is already on-par with the other frameworks. There are published benchmarks here for reference: [Robius Blog](https://robius.rs/blog/)

Let us know if you have any specific areas of concern regarding current performance, and we can definitely look into it.

**Comment by makepaddev** [comment]
The primary area we expect vulkan to be a win for is CPU time spent on the graphics API. We expect this to lower our cpu-per-frame time for the openGL vs vulkan code to be about a 50% win (or maybe faster). Consider that the openGL code for some of our examples is about 30-50% of the total cpu time we should be able to make it about 25% faster. However makepad is already so low on the CPU that we are mostly talking about battery efficiency and not really framerates here.

**Comment by guoguo338** [comment]
It's is so great! Makepad uses immediate mode for rendering, I can see it is very convenient to transfer those graphics algorithms codes which is CPU intensive and easy to be parallelized to GPU by compute shader, do you know what algorithms are suitable, especially for 2D rendering on mobile apps (not game).
